File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    while progresses:
        count = 0
        for index in range(0, len(progresses)):
            progresses[index] += speeds[index]
        while progresses and progresses[0] >= 100:
            count += 1
            progresses.pop(0)
            speeds.pop(0)
        if count != 0:
            answer.append(count)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var answer = [Int]()
    var progressesStack = progresses
    var speedStack = speeds
    while progressesStack.count != 0 {
        var count = 0
        for i in 0..<progressesStack.count {
            progressesStack[i] += speedStack[i]
        }
        while progressesStack.count != 0 && progressesStack[0] >= 100 {
            progressesStack.remove(at: 0)
            speedStack.remove(at: 0)
            count += 1
        }
        if count != 0 {
            answer.append(count)
        }

    }
    return answer
}